src/components/geolocation/GeolocationDirective.js

```diff
@@ -47,6 +57,23 @@
           }
         });
 
+        // Scope watchers
+        scope.$watch('tracking', function(tracking) {
```

Comment:
I think we don't need this watcher, Why don't we use the `change:heading' event as we do before (see line 137 )